File: LeetCode/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // Create a hash map, elements will be our key, indices will be our value
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
          // we are looking for the complement of each element that will equal to our target
          int complement = target - nums[i];
          // if our map contains the value of complement return the key for it and i in an array
          if(map.containsKey(complement))
            return new int[] {i, map.get(complement)};
          else
            // add our element and index to the map
            map.put(nums[i], i);
          }
        // we have not found the two sum and return {-1,-1}
        return new int[] {-1,-1};
    }
}

File: Random Code Challenges/twoSum.js
function twoSum(array, target){
  // Create a map
  let map = new Map();
  // iterate through the array
  for(let i = 0; i < array.length; i++){
    // set the complement for each element of the array
    let complement = target - array[i];
    // if our map has the complement return the index and the key
    if(map.has(complement))
      return [i,map.get(complement)];
    // if it is not in our map, add it
    else 
      map.set(array[i],i);
  }
  // exit and return -1,-1 if we have not found it
  return [-1,-1];
}
